tests/cases/fourslash/typeErrorAfterStringCompletionsInNestedCall.ts

```diff
@@ -25,7 +25,7 @@
 
 goTo.marker("1");
 edit.insert(`x`)
-verify.completions({ exact: ["MORNING", "LUNCH_TIME", "ALOHA"] });
+verify.completions({ exact: ["ALOHAx", "MORNING", "LUNCH_TIME", "ALOHA"] });
```

Comment:
This kind of makes sense since it is actively changing the inference to `TEvent` in `raise`, but the experience is not really desirable. I think VS Code showing you completion items that fully match your current token text is intentional and desirable in general, though. The reason this completion entry is bad is that the token being completed is "self-fulfilling," i.e. that text/type doesn't appear anywhere else. I fixed property assignments like that a while ago, so maybe a similar strategy could be used for string literals. I don't think it's a blocker for this PR, but I think it's worth investigating. https://github.com/microsoft/TypeScript/pull/35709

Reply:
I don't know, on second thought, if you change the token text to a substring of one of the valid options, like `ALO/**/`, it's going to be really annoying if the visible completions in VS Code are `ALO` first and `ALOHA` second.

Reply:
> I don't know, on second thought, if you change the token text to a substring of one of the valid options, like ALO/**/, it's going to be really annoying if the visible completions in VS Code are ALO first and ALOHA second.

This is not a new problem so I think it shouldn't be a blocker here ([TS playground](https://www.typescriptlang.org/play/?ts=5.5.0-dev.20240501#code/C4TwDgpgBA4gThCwCWA7A5gUQG4VcKAXgFgAoKKAHygCIBZAeQCUA5ASRZhrIupoBkAqiwDCACQD6AFTZ1M3clVoBBfgzHKaAbjJk0wCHABmAQwDG0JieQBnCMrMoA9qgYAjAFYRHAHik48YAA+KABvHihQSAAuWjhrO20IiFx8WP9U4B1SAF9dUgATbwAbEwQoIwBXVEdkFyh42wg-APwggApk7HTW4DIASlirJodnV09vYBbMoOyyIrNS8qqasagzBBMDOnMACzRmjMCOsxcjZHRY8MVzMZshhPtal3cvXyO27JzBqGwnZAKc1IGwgWwgOzM+1QzXgiBQGF6HWuFFudVQ9waj3aNFUDBo-QANGRvlogA)):
```ts
type GreetingEvent =
  | "MORNING"
  | "LUNCH_TIME"
  | "ALOHA";

interface RaiseActionObject<TEvent> {
  type: "raise";
  event: TEvent;
}

declare function raise<TEvent>(
  ev: TEvent
): RaiseActionObject<TEvent>;

declare function createMachine<TEvent>(config: {
  actions: RaiseActionObject<TEvent>;
}): void;

createMachine<GreetingEvent>({
  actions: raise("ALO"),
});
```

We can get those completions here:
<img width="625" alt="completions popup showing both ALO and ALOHA" src="https://github.com/microsoft/TypeScript/assets/9800850/b638a3c0-ad26-42c4-b800-422607913458">

All this PR does is reusing the *existing* algorithm in a new code path.

Reply:
The cases in the TS playground above and in the added fourslash test are different (and yet the same :P).

Strings used directly at argument positions and somewhere deeper within arguments (like as property values) hit 2 different codepaths. So what we observe on the TS playground above is a result of that other codepath and what this PR does is to port some logic from that other codepath to the other place. So **now** this fourslash test starts to behave like the TS playground above.